File: java/builder/src/main/OldCustomerProductBuilder.java
package main;

import java.math.BigDecimal;

public class OldCustomerProductBuilder extends ProductBuilder{

    Product product = new Product();

    @Override
    public void getProductData() {
        product.id = 1;
        product.categoryName = "teknoloji";
        product.name = "bilgsayar";
        product.unitPrice = BigDecimal.valueOf(30000);
    }

    @Override
    public void applyDiscount() {
        product.discountedPrice = product.unitPrice;
        product.discountApplied = false;
    }

    @Override
    public Product getProduct() {
        return product;
    }
}

File: csharp/Builder/OldCustomerProductBuilder.cs
namespace Builder;

public class OldCustomerProductBuilder : ProductBuilder
{
    ProductModel model = new ProductModel();

    public override void ApplyDiscount()
    {
        model.DiscountedPrice = model.UnitPrice;
        model.DiscountApplied = false;
    }

    public override void GetProductData()
    {
        model.Id = 1;
        model.CategoryName = "Teknoloji";
        model.Name = "bilgisayar";
        model.UnitPrice = 30000;
    }

    public override ProductModel GetProductModel()
    {
        return model;
    }
}